hr_holidays_leave_auto_approve/models/hr_leave.py

```diff
@@ -21,15 +21,17 @@ def create(self, values):
             HrLeave, self.with_context(
                 tracking_disable=tracking_disable)
             ).create(values)
-        if res.can_approve and res.holiday_status_id.auto_approve:
-            res.action_approve()
+        policy = res.holiday_status_id.auto_approve_policy
+        if (res.can_approve and policy == 'hr') or policy == 'all':
```

Comment:
For further extensions, better extract this to a separate hook method, e.g. "apply_auto_approve_policy"